### DISCUSSION TITLE
Usage of vvveb as html editor

### DISCUSSION THREAD

**Discussion by Smanst3r** [givanz/VvvebJs#325] [Q&A]
Hi, the demo https://www.vvveb.com/vvvebjs/editor.html looks awesome. 
But I couldn't find any information on how to install vvvebJs and use it without the demo folder that comes across the downloaded archive.
![Screenshot 2023-09-28 17:28:52](https://github.com/givanz/VvvebJs/assets/8392953/c517976a-fe2f-4f2b-a2ce-6856383fc923)
Is there any information available on how to use only the blocks highlighted in green in the screenshot? (for example: _should I create a custom builder.js file to include only the necessary components? Is the editor.html file required, or is it intended solely for demonstration purposes, and so on..._)

My goal is as follows:
1. Pass HTML string into the vvvebJs editor.
2. Perform various HTML manipulations, (_such as dragging blocks, editing text, adjusting colors, managing links, inserting images, modifying styles, etc..._)
3. Retrieve the result of my modifications in HTML format.
Is this achievable? Sincerely

**Comment by givanz** [comment]
Hi

Thank you.

To remove the file manager delete the following html code 
https://github.com/givanz/VvvebJs/blob/master/editor.html#L119-L140

and comment or delete file manager code 

https://github.com/givanz/VvvebJs/blob/master/editor.html#L1729 
https://github.com/givanz/VvvebJs/blob/master/editor.html#L1733-L1734


To load the content from a string you need to remove the code that loads the page from a url/file from https://github.com/givanz/VvvebJs/blob/master/editor.html#L1724-L1726

and replace it with 

```js
	Vvveb.Builder.init("about:blank", function() {
		//load code after page is loaded here
		Vvveb.Builder.setHtml("<h1>Hello world!</h1>");
	});
```	


You can get the html code of the edited page with

```js
Vvveb.Builder.getHtml();
```

Or you can replace the save code to save to database instead of file https://github.com/givanz/VvvebJs/blob/master/save.php#L124

`editor.html` is required, it includes all the editor elements and input templates.

It's easier to maintain and customize the elements in editor.html directly than to have them created from javascript and add extra code and complexity.

**Comment by hayden-t** [comment]
Thank you for this ! just fyi some of the line number references dont line up now since master source code changes.
